src/dev_cron.erl

```diff
@@ -1,75 +1,301 @@
+	ReqSingleton = Req#{ <<"path">> => CronPath },
+	?event({cron_every_worker_executing, {path, CronPath}, {req_id, hb_message:id(Req, all)}}),
+	try
+		Result = dev_meta:handle(Opts, ReqSingleton),
+		?event({cron_every_worker_executed, {path, CronPath}, {result, Result}})
+	catch
+		Class:Reason:Stacktrace ->
+			?event({cron_every_worker_error, {path, CronPath}, {error, Class, Reason, Stacktrace}}),
+			timer:sleep(IntervalMillis),
+			every_worker_loop(CronPath, Req, Opts, IntervalMillis)
+	end,
+	timer:sleep(IntervalMillis),
+	every_worker_loop(CronPath, Req, Opts, IntervalMillis).
+
+%% @doc Parse a time string into milliseconds.
 parse_time(BinString) ->
-    [AmountStr, UnitStr] = binary:split(BinString, <<"-">>),
-    Amount = binary_to_integer(AmountStr),
-    Unit = string:lowercase(binary_to_list(UnitStr)),
-    case Unit of
-        "millisecond" ++ _ -> Amount;
-        "second" ++ _ -> Amount * 1000;
-        "minute" ++ _ -> Amount * 60 * 1000;
-        "hour" ++ _ -> Amount * 60 * 60 * 1000;
-        "day" ++ _ -> Amount * 24 * 60 * 60 * 1000;
-        _ -> throw({error, invalid_time_unit, UnitStr})
-    end.
-
-execute(_M, State = #{ <<"cron">> := inactive }) ->
-    {ok, State};
-execute(M, State = #{ <<"pass">> := 1, <<"cron">> := #state { last_run = undefined } }) ->
-    {ok, State#{ <<"cron">> := #state { last_run = timestamp(M) } }};
-execute(Message, State = #{ <<"pass">> := 1, <<"cron">> := #state { time = MilliSecs, last_run = LastRun }, <<"schedule">> := Sched }) ->
-    case timestamp(Message) - LastRun of
-        Time when Time > MilliSecs ->
-            NextCronMsg = create_cron(State, CronTime = timestamp(Message) + MilliSecs),
-            {pass,
-                State#{
-                    <<"cron">> := #state { last_run = CronTime },
-                    <<"schedule">> := [NextCronMsg | Sched]
-                }
-            };
-        _ ->
-            {ok, State}
-    end;
-execute(_, S) ->
-    {ok, S}.
-
-timestamp(M) ->
-    % TODO: Process this properly
-    case lists:keyfind(<<"timestamp">>, 1, M#tx.tags) of
-        {<<"timestamp">>, TSBin} ->
-            list_to_integer(binary_to_list(TSBin));
-        false ->
-            0
-    end.
-
-create_cron(_State, CronTime) ->
-    #tx{
-        tags = [
-            {<<"Action">>, <<"Cron">>},
-            {<<"Timestamp">>, list_to_binary(integer_to_list(CronTime))}
-        ]
-    }.
-
-uses() -> all.
\ No newline at end of file
+	[AmountStr, UnitStr] = binary:split(BinString, <<"-">>),
+	Amount = binary_to_integer(AmountStr),
+	Unit = string:lowercase(binary_to_list(UnitStr)),
+	case Unit of
+		"millisecond" ++ _ -> Amount;
+		"second" ++ _ -> Amount * 1000;
+		"minute" ++ _ -> Amount * 60 * 1000;
+		"hour" ++ _ -> Amount * 60 * 60 * 1000;
+		"day" ++ _ -> Amount * 24 * 60 * 60 * 1000;
+		_ -> throw({error, invalid_time_unit, UnitStr})
+	end.
+
+%%% Tests
+
+stop_once_test() ->
+	% Start a new node
+	Node = hb_http_server:start_node(),
+	
```

Comment:
Existing style of the codebase is no whole-line whitespace -- general principle is if there is a logical separation, write a whole-line comment for what the below section does.